server/src/main/java/org/candlepin/model/ProductContent.java

```diff
@@ -144,6 +144,22 @@ public int hashCode() {
     }
 
     /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
+        int hash = 17;
+
+        hash = 7 * hash + (this.content != null ? this.content.getEntityVersion() : 0);
```

Comment:
I imagine it would, but now you're hashing a hash just to use the HCB. You still have to check for null, so the only logic it saves us is the check on enabled.